engine.py

```diff
@@ -61,9 +61,47 @@ def init_engine(self):
         version_str = Application.version()
 
         try:
+            # Attempt getting the year version from product name version info block
+            try:
+                import win32api
+                import re
+
+                # Parses the Windows FileInfo block to extract the ProductName field.
+                application_full_path = Application.FullName
+
+                # Need to query the version info block based on file's locale
+                language, codepage = win32api.GetFileVersionInfo(application_full_path, "\\VarFileInfo\\Translation")[0]
+                stringFileInfo = "\\StringFileInfo\\%04X%04X\\%s" % (language, codepage, "ProductName")
+                product_name = win32api.GetFileVersionInfo(application_full_path, stringFileInfo)
+                metric_logged_version = re.sub("[^0-9]*", "", product_name)
+                self.logger.debug("Polled year version (%s) from 'ProductName' file's version info block." % (str(metric_logged_version)))
+
+            except:
+                # On ANY exception try relying on the Softimage API which
+                # needs to be maintained manually
+                try:
+                    version_parts = version_str.split(".")
+                    if version_parts and version_parts[0].isnumeric():
+                        version_major = int(version_parts[0])
+                        # At least since 2011 the major version seems to be incremented each year
+                        # 13 = 2015, 11 = 2013, others are unverified
+                        if version_major == 11:
+                            metric_logged_version = "2013"
+                        elif version_major == 13:
+                            metric_logged_version = "2015"
+                        else:
+                            raise Exception("Unexpected Major Version")
+
+                        self.logger.debug("Polled year version (%s) based 'version_major' version." % (
+                        str(metric_logged_version)))
+                except:
+                    # Worst case fallback, just use whatever was returned by the Soft Image API
+                    self.logger.debug("Polled version (%s) from SoftImage API." % (str(version_str)))
+                    metric_logged_version = version_str
+
             # Create a _host_info variable that we can update so later usage of
             # the `host_info` property can benefit having the updated information.
-            self._host_info = {"name": "Softimage", "version": version_str}
+            self._host_info = {"name": Application.Name, "version": metric_logged_version}
```

Comment:
Why not use `Application.Name` above where ever product name is required? (in the log messages)